src/controller/cmcd-controller.ts

```diff
@@ -164,24 +165,24 @@ export default class CMCDController implements ComponentAPI {
 
     // TODO: Implement rtp, nrr, nor, dl
 
-    if (this.useHeaders) {
-      const headers = CMCDController.toHeaders(data);
-      if (!Object.keys(headers).length) {
-        return;
-      }
+    const { includeKeys } = this;
+    if (includeKeys) {
+      data = Object.keys(data)
+        .filter((key) => includeKeys.includes(key))
+        .reduce((acc, key) => {
+          acc[key] = data[key];
+          return acc;
+        }, {});
```

Comment:
Can you do the filtering in the reduce callback?

```suggestion
      data = Object.keys(data).reduce((acc, key) => {
        includeKeys.includes(key) && (acc[key] = data[key]);
        return acc;
      }, {});
```